src/terminator/intrinsic.rs

```diff
@@ -426,10 +426,7 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
                     let (sized_size, sized_align) = match *layout {
                         ty::layout::Layout::Univariant { ref variant, .. } => {
-                            // The offset of the start of the last field gives the size of the
-                            // sized part of the type.
-                            let size = variant.offsets.last().map_or(0, |f| f.bytes());
-                            (size, variant.align.abi())
+                            (variant.offsets.last().map_or(0, |o| o.bytes()), variant.align.abi())
```

Comment:
You should probably not call `.abi()` until the last moment if you can avoid it?